diff -Naur rpm/scripts/oech_logrotate.sh oech/scripts/oech_logrotate.sh
--- rpm/scripts/oech_logrotate.sh	2022-08-29 19:51:34.000000000 +0800
+++ oech/scripts/oech_logrotate.sh	2022-09-05 17:15:26.557831648 +0800
@@ -42,7 +42,8 @@
 }
 
 function stop_logrotate() {
-    ps -ef | grep oech_logrotate.sh | grep -v grep | awk '{print $2}' | xargs kill -9 >/dev/null 2>&1
+    ps -ef | grep oech_logrotate.sh | grep -v grep | awk '{print $2}' | xargs killall >/dev/null 2>&1
+    return 0
 }
 
 function main() {
@@ -57,4 +58,4 @@
     fi
 }
 
-main "$@"
\ No newline at end of file
+main "$@"
