diff -Naur rpm/tests/disk/common.py oech/tests/disk/common.py
--- rpm/tests/disk/common.py	2022-10-26 19:50:53.857748470 +0800
+++ oech/tests/disk/common.py	2022-10-26 19:38:37.545748470 +0800
@@ -39,7 +39,7 @@
     command.run_cmd("cat /proc/mdstat")
 
 
-def get_disk(logger, command, config_data, pci_num):
+def get_disk(logger, command, config_data, pci_num=""):
     """
     Get disk info
     """
diff -Naur rpm/tests/disk/disk.py oech/tests/disk/disk.py
--- rpm/tests/disk/disk.py	2022-10-26 19:50:53.857748470 +0800
+++ oech/tests/disk/disk.py	2022-10-26 19:38:37.545748470 +0800
@@ -42,7 +42,7 @@
         """
         Start test
         """
-        self.disks = get_disk(self.logger, self.command, self.config_data, self.pci_num)
+        self.disks = get_disk(self.logger, self.command, self.config_data)
         if len(self.disks) == 0:
             self.logger.error("No suite disk found to test.")
             return False
