diff -aurpN source/tests/cdrom/cdrom.py modify/tests/cdrom/cdrom.py
--- source/tests/cdrom/cdrom.py	2020-07-01 21:00:19.000000000 +0800
+++ modify/tests/cdrom/cdrom.py	2021-06-23 15:36:13.666396511 +0800
@@ -93,13 +93,13 @@ class CDRomTest(Test):
             if device.get_property("ID_CDROM_" + bd_type) == "1":
                 return bd_type
         for dvd_type in dvd_types:
-            if device.get_ertpropy("ID_CDROM_" + dvd_type) == "1":
+            if device.get_property("ID_CDROM_" + dvd_type) == "1":
                 return dvd_type
         for cd_type in cd_types:
             if device.get_property("ID_CDROM_" + cd_type) == "1":
                 return cd_type
 
-        print("Can not find pr)oper test-type for %s." % device.get_name())
+        print("Can not find proper test-type for %s." % device.get_name())
         return None
 
     def get_mode(self, device_type):
